sklearn/linear_model/_base.py

```diff
@@ -179,9 +180,12 @@ def _preprocess_data(X, y, fit_intercept, normalize=False, copy=True,
 # Currently, the fact that sag implements its own way to deal with
 # sample_weight makes the refactoring tricky.
 
-def _rescale_data(X, y, sample_weight):
+def _rescale_data(X, y, sample_weight, order='C'):
```

Comment:
Do we actually always need the order? I guess some of the algorithms work either way. Maybe add keep `order=None` by default which wouldn't constrain order.